### ISSUE TITLE
MSIntue SDK iOS app not redirecting after login and enrollment

### ISSUE THREAD

**Issue by MikeP0911** [msintuneappsdk/intune-app-sdk-xamarin#56]
After 14.3.5 Intune version, IntuneMAMEnrollmentManager.Instance.LoginAndEnrollAccount(emailText); is not working.
Enrollment has not taken place.

LoginAndEnrollAccount should do authentication
Expected behavior:
Authentication should be performed

Intune App SDK for iOS (please complete the following information):

What version of the Intune SDK are you using? Are you using the latest version? Yes 17.0.0
What platform is your app based in (native, Xamarin based, Cordova, etc)? Xamarin based
The Intune Login is not happening and the screen is freezing.

Authentication is handled by Intune SDK and not by ADAL or MSAL.
https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-xamarin
Developed with the help of the above link.

**Comment by MikeP0911** [comment]
IntuneMAMEnrollmentManager.Instance.LoginAndEnrollAccount(emailText)

is not prompting the user for user name and password when Microsoft.Intune.MAM.Xamarin.iOS version is 17.1.1

it prompts for username and password when Microsoft.Intune.MAM.Xamarin.iOS version is 14.3.5

**Comment by abasore** [comment]
Hi @MikeP0911, can you confirm if enrollment is succeeding for the user and policy is being applied? Its possible that this expected. The SDK might be using cached data to skip the redirecting the user for the login.

**Comment by MikeP0911** [comment]
@abasore The enrollment is not succeeded and the policy is not applied. 
When I click the Intune login button there is no response in recent updated Intune SDK version.
But when I use the 14.3.5 version it asks for login credentials and login is carried out successfully and policies are applied.

**Comment by MikeP0911** [comment]
@Kyle-Reis We are using Xamarin Bindings and the authentication is taken care by Intune SDK.
Please suggest. There are lot of users affected by it.

**Comment by MikeP0911** [comment]
https://user-images.githubusercontent.com/117801555/210716817-6a6ca74a-735a-40c7-bc4e-ad4137f21268.MP4

https://user-images.githubusercontent.com/117801555/210716824-4b7740f6-4053-4275-af69-8b3f8fb0259a.MP4

**Comment by MikeP0911** [comment]
Please refer the video for the working condition of the Intune SDK.
When the Intune SDK is 14.3.5 version, it asks for the credentials. If login is successful then the app is ready to use smoothly.
When the Intune SDK is 17.1.1 version, by clicking Intune login button nothing happens.

**Comment by Kyle-Reis** [comment]
Hi @MikeP0911 - Version 14.3.5 of the Intune Xamarin.iOS bindings happens to be the last version of the bindings which used the native version Objective-C version of the deprecated ADAL authentication library. After that, the switch over to MSAL was made, as mentioned in the notes for the  [release which followed it](https://github.com/msintuneappsdk/intune-app-sdk-xamarin/releases/tag/9%2F13%2F2021) (14.7.0). With the switch to MSAL, applications were required to update the redirect URI used for Intune enrollment to match the [MSAL-required format](https://github.com/AzureAD/microsoft-authentication-library-for-objc#configuring-msal). Does your application configure any of the ADAL settings mentioned [here](https://learn.microsoft.com/mem/intune/developer/app-sdk-ios#configure-msal-settings-for-the-intune-app-sdk)? If so, could you ensure that the redirect URI configured is in the required format? Also make sure that this redirect URI is specified in the AAD app registration for your application at https://portal.azure.com/#view/Microsoft_AAD_RegisteredApps/ApplicationsListBlade.

**Comment by MikeP0911** [comment]
@Kyle-Reis Thank you for your response. We do not use any of the ADAL settings in our project. We have not used any of the ADAL settings．In this case what is our next step. How could we make our application work with the recent Intune Library.

**Comment by Kyle-Reis** [comment]
@MikeP0911 - When linking to the latest Intune release and building the app with Visual Studio, is there a build warning about running the IntuneMAMConfigurator?

**Comment by MikeP0911** [comment]
@MikeP0911 - below are the warnings

**when I use 14.3.5 the warning is** 
warning: object file (/Users/user1/Library/Caches/Xamarin/mtbs/builds/ZipExtractor.iOS/e1ba4871388be6b2c5f666c192f9b1dfbc109a8853ad6b5fc47b2e1b7b9d61cc/obj/iPhone/A1_0_LicenseDebug/device-builds/iphone12.1-16.2/mtouch-cache/libIntuneMAM.a(libIntuneMAM.a-arm64-master.o)) was built for newer iOS version (12.2) than being linked (12.0)

**when I use the 17.1.1 the warning is** 

warning: dylib (/Users/user1/Library/Caches/Xamarin/mtbs/builds/ZipExtractor.iOS/e1ba4871388be6b2c5f666c192f9b1dfbc109a8853ad6b5fc47b2e1b7b9d61cc/obj/iPhone/A1_0_LicenseDebug/device-builds/iphone12.1-16.2/mtouch-cache/IntuneMAMSwift.framework/IntuneMAMSwift) was built for newer iOS version (13.0) than being linked (12.0)

warning: dylib (/Users/user1/Library/Caches/Xamarin/mtbs/builds/ZipExtractor.iOS/e1ba4871388be6b2c5f666c192f9b1dfbc109a8853ad6b5fc47b2e1b7b9d61cc/obj/iPhone/A1_0_LicenseDebug/device-builds/iphone12.1-16.2/mtouch-cache/IntuneMAMSwiftStub.framework/IntuneMAMSwiftStub) was built for newer iOS version (13.0) than being linked (12.0)

**Comment by MikeP0911** [comment]
@Kyle-Reis Below are the warnings

Severity Code Description Project File Line Suppression State
Warning NETSDK1138 The target framework 'net5.0' is out of support and will not receive security updates in the future. Please refer to https://aka.ms/dotnet-core-support for more information about the support policy. SharpCompress C:\Program Files\dotnet\sdk\7.0.102\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.EolTargetFrameworks.targets 28

Warning Found conflicts between different versions of "Microsoft.CSharp" that could not be resolved.
There was a conflict between "Microsoft.CSharp, Version=2.0.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" and "Microsoft.CSharp, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
"Microsoft.CSharp, Version=2.0.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was chosen because it was primary and "Microsoft.CSharp, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was not.
References which depend on "Microsoft.CSharp, Version=2.0.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" [C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\Xamarin.iOS\v1.0\Microsoft.CSharp.dll].
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\Xamarin.iOS\v1.0\Microsoft.CSharp.dll
Project file item includes which caused reference "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\Xamarin.iOS\v1.0\Microsoft.CSharp.dll".
Microsoft.CSharp
References which depend on "Microsoft.CSharp, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" [].
C:\Users\veluchamy.rathika.nuget\packages\csvhelper\27.2.1\lib\netstandard2.1\CsvHelper.dll
Project file item includes which caused reference "C:\Users\veluchamy.rathika.nuget\packages\csvhelper\27.2.1\lib\netstandard2.1\CsvHelper.dll".
C:\Users\veluchamy.rathika.nuget\packages\csvhelper\27.2.1\lib\netstandard2.1\CsvHelper.dll
C:\TFS Source\LicenseKanri\ZipExtractor\ZipExtractor\ZipExtractor\bin\A1_0_LicenseDebug\netstandard2.0\ZipExtractor.dll
C:\Users\veluchamy.rathika.nuget\packages\microsoft.identitymodel.clients.activedirectory\5.2.9\lib\xamarinios10\Microsoft.IdentityModel.Clients.ActiveDirectory.dll
Project file item includes which caused reference "C:\Users\veluchamy.rathika.nuget\packages\microsoft.identitymodel.clients.activedirectory\5.2.9\lib\xamarinios10\Microsoft.IdentityModel.Clients.ActiveDirectory.dll".
C:\Users\veluchamy.rathika.nuget\packages\microsoft.identitymodel.clients.activedirectory\5.2.9\lib\xamarinios10\Microsoft.IdentityModel.Clients.ActiveDirectory.dll ZipExtractor.iOS

Warning The app icon set "AppIcons_ZipExtractor" has 9 unassigned children. ZipExtractor.iOS

Warning Debugging symbol file for '/Users/user1/Library/Caches/Xamarin/mtbs/builds/ZipExtractor.iOS/abaa0656008ad72677ac56a932594932cb43a636387f36971cf878185b335679/bin/iPhone/A1_0_LicenseDebug/device-builds/ipad6.11-16.2/ZipExtractor.dll' is not valid and was ignored. ZipExtractor.iOS

Warning Debugging symbol file for '/Users/user1/Library/Caches/Xamarin/mtbs/builds/ZipExtractor.iOS/abaa0656008ad72677ac56a932594932cb43a636387f36971cf878185b335679/bin/iPhone/A1_0_LicenseDebug/device-builds/ipad6.11-16.2/ZipExtractor.dll' is not valid and was ignored. ZipExtractor.iOS
Warning The project 'SharpCompress' cannot be referenced. The referenced project is targeted to a different framework family (.NETFramework) ZipExtractor.iOS

Warning NU1605 Detected package downgrade: Xamarin.Forms from 5.0.0.2545 to 5.0.0.2401. Reference the package directly from the project to select a different version.
ZipExtractoriOS -> ZipExtractor -> Xamarin.Forms (>= 5.0.0.2545)
ZipExtractoriOS -> Xamarin.Forms (>= 5.0.0.2401) 1

Warning NU1605 Detected package downgrade: CsvHelper from 27.2.1 to 21.2.0. Reference the package directly from the project to select a different version.
ZipExtractor.AndroidTest -> ZipExtractor -> CsvHelper (>= 27.2.1)
ZipExtractor.AndroidTest -> CsvHelper (>= 21.2.0) 1

Warning NU1605 Detected package downgrade: Newtonsoft.Json from 13.0.1 to 12.0.3. Reference the package directly from the project to select a different version.
ZipExtractor.Android -> ZipExtractor -> Newtonsoft.Json (>= 13.0.1)
ZipExtractor.Android -> Newtonsoft.Json (>= 12.0.3) 1

Warning NU1605 Detected package downgrade: Xamarin.Forms from 5.0.0.2545 to 4.4.0.991757. Reference the package directly from the project to select a different version.
ZipExtractor.Android -> ZipExtractor -> Xamarin.Forms (>= 5.0.0.2545)
ZipExtractor.Android -> Xamarin.Forms (>= 4.4.0.991757) 1

Warning NU1605 Detected package downgrade: Xamarin.Essentials from 1.7.4 to 1.6.1. Reference the package directly from the project to select a different version.
ZipExtractor.UWP -> ZipExtractor -> Xamarin.Essentials (>= 1.7.4)
ZipExtractor.UWP -> Xamarin.Essentials (>= 1.6.1) 1

Warning NU1605 Detected package downgrade: NLog.Config from 4.7.15 to 4.7.7. Reference the package directly from the project to select a different version.
ZipExtractor.Android -> ZipExtractor -> NLog.Config (>= 4.7.15)
ZipExtractor.Android -> NLog.Config (>= 4.7.7) 1

Warning NU1605 Detected package downgrade: Newtonsoft.Json from 13.0.1 to 12.0.3. Reference the package directly from the project to select a different version.
ZipExtractor.AndroidTest -> ZipExtractor -> Newtonsoft.Json (>= 13.0.1)
ZipExtractor.AndroidTest -> Newtonsoft.Json (>= 12.0.3) 1

Warning NU1605 Detected package downgrade: Xamarin.Essentials from 1.7.4 to 1.7.3. Reference the package directly from the project to select a different version.
ZipExtractoriOS -> ZipExtractor -> Xamarin.Essentials (>= 1.7.4)
ZipExtractoriOS -> Xamarin.Essentials (>= 1.7.3) 1

**Comment by MikeP0911** [comment]
Hi @Kyle-Reis ,
We are not able to see any warning message related to intune SDK.
My application uses Xamarin Forms 5.0.0.2545
Xamarin Essentials 1.7.4
Is it not working due to any dependency issues.
Thank you in advance

**Comment by MikeP0911** [comment]
Hi Mike, 
I am sending the file to your mail id.
Please check and revert.
Thank you

**Comment by MikeP0911** [comment]
Hi Kyle,

I have updated Intune SDK to 17.4.2.
Building with Xcode 14.3.
MAC OS Ventura 13.3.1 (a) 
Xamarin.Forms 5.0.0.2578
Xamarin.Essentials 1.7.5

We use the default Intune Authentication.
When the Intune login button is pressed, the authentication is not completed.
Is there any extra keys that has to be set when using the latest Intune SDK.
We are updating from Intune SDK 14.3.5. Hence is there something we are missing.
We have registered our app in Azure Active directory. But we have not used it anywhere in our application.
Is that affecting the login process.

Help is much appreciated.
